articles/list-transformations-js/index.md

```diff
@@ -0,0 +1,357 @@
+﻿
+## List transformations in JavaScript
+
+In this tutorial, we will be looking at creating new objects and arrays from existing ones. We will look at creating nested data structures, looping and destructuring them. We will also look at transforming data structures by passing them through functions. 
+
+### Prerequisites
+
+To follow through this tutorial, you are required to have at least some knowledge in JavaScript [objects and arrays](https://www.section.io/engineering-education/object-arrays-destructuring/).
+
+By the end of this tutorial, you will be able to:
+- Nest objects and arrays.
+- Loop through the nested data structures.
+- Destructure nested data structures.
+- Transform objects and arrays using underscore.js.
+
+## Step 1 -- Nesting objects and arrays
+
+Data structures can be nested to avoid the declaration of many variables. Multiple objects and/or arrays can be held by a single variable and can be accessed using [dot notation and bracket notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors).
+```javascript
+const cars = {
+    'brands': [
+        {
+            name: "Ford",
+            origin: "USA"
+        },
+        {
+            name: "Toyota",
+            origin: "Japan"
+        }
+    ]
+}
+```
+We create an object `cars` and nest an array `brands` inside it. We also add 2 objects inside the `brands` array.
+Nesting can also be done using either the dot notation or square bracket notation where necessary.
```

Comment:
"... where applicable." would be appropriate in this case as opposed to "... where necessary."
To indicate that either of the two notations can be used where it makes sense to use it.